dnscache/storage.go

```diff
@@ -0,0 +1,94 @@
+package dnscache
+
+import (
+	"net"
+	"reflect"
+	"time"
+
+	"fmt"
+
+	"github.com/pmylund/go-cache"
+)
+
+// DnsCacheItem represents single record in cache
+type DnsCacheItem struct {
```

Comment:
a little confusing the purpose of this struct, would be equal to use just `type DnsCacheItem []string` or store in cache just `string`

Reply:
actually just `string` might not work in case we want to know that this domain has several IPs and build some logic on top of it, i.e. always do real DNS request foor those hostnames in cache

Reply:
Let's keep it like this for now.